src/pages/api/arbeidssoker/perioder.ts

```diff
@@ -1,75 +1,51 @@
-import { NextApiHandler } from "next";
+import { NextApiHandler, NextApiRequest, NextApiResponse } from "next";
 import { getSession } from "../../../lib/auth.utils";
 import { v4 as uuid } from "uuid";
-import path from "path";
-import { decodeJwt } from "@navikt/dp-auth";
+import { formatISO } from "date-fns";
+import { veilarbAudience } from "../../../lib/audience";
 
 export type Arbeidssøkerperiode = {
   fraOgMedDato: string;
   tilOgMedDato: string;
 };
 
-const periodeFormatter = new Intl.DateTimeFormat("no", {
-  year: "numeric",
-  month: "2-digit",
-  day: "2-digit",
-});
-
-function formaterDato(date: Date) {
-  return periodeFormatter.format(date).split(".").reverse().join("-");
-}
-
-export const leggTilQueries = (fnr: string): string => {
-  const idag = new Date();
-  const query = new URLSearchParams();
-  query.append("fnr", fnr);
-  query.append("fraOgMed", formaterDato(idag));
-
-  return `?${query.toString()}`;
-};
-
 const perioderHandler: NextApiHandler<Arbeidssøkerperiode[]> = async (
-  req,
-  res
+  req: NextApiRequest,
+  res: NextApiResponse
 ) => {
-  const session = await getSession(req);
+  const callId = uuid();
 
-  if (!session.token) return res.status(401).end();
-  const payload = decodeJwt(session.token);
+  try {
+    const { token, apiToken } = await getSession(req);
 
-  const callId = uuid();
-  const url = new URL(process.env.VEILARBPROXY_URL);
-  url.pathname = path.join(url.pathname, "/api/arbeidssoker/perioder");
-  url.search = leggTilQueries(<string>payload.pid);
+    if (!token) {
+      return res.status(401).end();
+    }
 
-  console.log(
-    `Henter arbeidssøkerperioder fra veilarbregistrering (callId: ${callId})`
-  );
+    const onBehalfOfToken = await apiToken(veilarbAudience);
 
-  try {
-    const { arbeidssokerperioder } = await fetch(url.toString(), {
+    const today = formatISO(new Date(), { representation: "date" });
+    const url = `${process.env.VEILARBPROXY_URL}/api/arbeidssoker/perioder/niva3?fraOgMed=${today}`;
```

Comment:
Husker ikke helt hva de forskjellige nivåer er, men trenger vi å spesifisere nivå her?

Reply:
Vi trenger å spesifisere `niva3` for å bruke det nye endepunktet som benytter tokenX. Dette endepunktet fungerer kun for personbruker (antar dette er nivået du refererer til?), henter fødselsnummer ut av tokenet, og returnerer akkurat det samme som endepunktet vi bruker i dag. 😊